src/main/java/org/apache/commons/imaging/common/BasicCParser.java

```diff
@@ -417,4 +341,103 @@ public static void unescapeString(final StringBuilder stringBuilder, final Strin
                     + "unterminated escape sequence found in string");
         }
     }
+
+    private static int appendHex(int i, final StringBuilder stringBuilder, final String string)
```

Comment:
Could you add some javadoc to these methods too, please, @snumlautoken ? The names are self-explanatory, but might still be helpful to be explicit about it, and also mention the use of the parameters.

Reply:
This has been added!